src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemAlignmentUtils.java

```diff
@@ -57,9 +67,15 @@ public static SAMRecord applyAlignment( final String readName, final byte[] base
                         }
                     }
                 }
-                bases = Arrays.copyOfRange(bases, alignment.getSeqStart(), alignment.getSeqEnd());
+            }
+            if (tmpCigar.getFirstCigarElement().getOperator() == CigarOperator.H ||
+                    tmpCigar.getLastCigarElement().getOperator() == CigarOperator.H) {
+                final int leftHardClip = tmpCigar.getFirstCigarElement().getOperator() == CigarOperator.H ? tmpCigar.getFirstCigarElement().getLength() : 0;
```

Comment:
Do ```getSeqStart/End``` return the base index range in the read that is actually aligned to the ref (i.e. involving first M and last M cigar elements) or that plus the soft-clipped range? I suspected is the former thus may change. Otherwise, happy to revert to that code.